dataactvalidator/interfaces/interfaceHolder.py

```diff
@@ -8,10 +9,17 @@ class InterfaceHolder:
 
     def __init__(self):
         """ Create the interfaces """
-        self.jobDb = ValidatorJobTrackerInterface()
-        self.errorDb = ValidatorErrorInterface()
-        self.stagingDb = ValidatorStagingInterface()
-        self.validationDb = ValidatorValidationInterface()
+        if BaseInterface.interfaces is None:
+            self.jobDb = ValidatorJobTrackerInterface()
+            self.errorDb = ValidatorErrorInterface()
+            self.stagingDb = ValidatorStagingInterface()
+            self.validationDb = ValidatorValidationInterface()
+            BaseInterface.interfaces = self
```

Comment:
> any time a new thread starts it needs to instantiate a separate InterfaceHolder to avoid conflicts in the session and connection objects.

Gotcha. That's wading much deeper than I anticipated. At some point, we'll probably want to replace these with `flask.g`: http://flask.pocoo.org/docs/0.11/api/#application-globals, but that can can be kicked down the road a bit.